frontend/src/metabase/common/components/FilterPicker/CoordinateFilterPicker/CoordinateColumnPicker/CoordinateColumnPicker.tsx

```diff
@@ -0,0 +1,46 @@
+import { useMemo, useState } from "react";
+import { Select, Stack } from "metabase/ui";
+import type * as Lib from "metabase-lib";
+import { getColumnOptions, getColumnPlaceholder } from "./utils";
+
+interface CoordinateColumnPickerProps {
+  query: Lib.Query;
+  stageIndex: number;
+  column: Lib.ColumnMetadata;
+  availableColumns: Lib.ColumnMetadata[];
+  onChange: (secondColumn: Lib.ColumnMetadata) => void;
+}
+
+export function CoordinateColumnPicker({
+  query,
+  stageIndex,
+  column,
+  availableColumns,
+  onChange,
+}: CoordinateColumnPickerProps) {
+  const options = useMemo(() => {
+    return getColumnOptions(query, stageIndex, availableColumns);
+  }, [query, stageIndex, availableColumns]);
+
+  const [value, setValue] = useState<string | null>(null);
```

Comment:
Previously initial state was not `null`

Reply:
It used an unstable hack to compute the initial state. Currently there is no correct way to do it, so I need to keep it as it is.

Reply:
Set the initial value